src/ndt-client.js

```diff
@@ -98,6 +115,64 @@ NDTjs.prototype.checkEnvironmentSupport = function() {
 };
 
 /**
+ * A generic login creation system for NDT.
+ *
+ * @param {Number} desiredTests The types of tests requested from the server
+ *  signaled based on a bitwise operation of the test ids.
+ * @returns {Uint8Array} An array of bytes suitable for sending on a binary
+ *  websocket.
+ */
+
+NDTjs.prototype.makeNDTLogin = function(desiredTests) {
+  var messageBody = '{ "msg": "v3.5.5", "tests": "' +
+      String(desiredTests | 16) + '" }';
+
+  return this.makeNDTMessageArray(this.constants.messageType.MSG_EXTENDED_LOGIN,
+      messageBody);
+};
+
+/**
+ * A generic message creation system for NDT.
+ *
+ * @param {Number} messageType The type of message according to NDT's
+ *  specification.
+ * @param {String} messageContent The message to send the server.
+ * @returns {Uint8Array} An array of bytes suitable for sending on a binary
+ *  websocket.
+ */
+
+NDTjs.prototype.makeNDTMessage = function(messageType, messageContent) {
+   var messageBody = '{ "msg": "' + messageContent + '" }';
```

Comment:
(JavaScript n00b question)
It seems surprising that we'd want to construct our own JSON by hand in JavaScript, the land of JSON. Could we do something like this?

`JSON.stringify({"msg": messageContent})`